### PR TITLE
obs_base: add tests for Butler.datasetExists with composites

### PR DESCRIPTION
**Pull request by n8pease**

### CODE REVIEW

File: tests/testComposite.py [hunk 1]
```diff
@@ -134,6 +134,23 @@ def testDottedDatasetType(self):
         verObjB = verificationButler.get('basicObject2', {'name': 'bar'})
         self.assertEqual(verObjB, componentObjB)
 
+    def testDatasetExists(self):
+        """Verify that Butler.datasetExists returns true for a composite dataset whose components exist."""
+        butler = dafPersist.Butler(inputs=self.firstRepoPath)
+        self.assertTrue(butler.datasetExists('basicPair', dataId={'id': 'foo', 'name': 'bar'}))
```
Comment (kennywlo) [inline tests/testComposite.py]: PEP8: line above too long.

File: tests/testComposite.py [hunk 2]
```diff
@@ -134,6 +134,23 @@ def testDottedDatasetType(self):
...
+        """Verify that Butler.datasetExists returns false for a composite dataset where some of the
+        components do not exist."""
+        repoPath = os.path.join(self.testData, 'repo')
+        repoArgs = dafPersist.RepositoryArgs(root=repoPath, policy=self.policy,
+                                             mapper='lsst.obs.base.test.CompositeMapper')
+
+        butler = dafPersist.Butler(outputs=repoArgs)
+        self.objA = dpTest.TestObject("abc")
+        butler.put(self.objA, 'basicObject1', dataId={'id': 'foo'})
+        self.assertFalse(butler.datasetExists('basicPair', dataId={'id': 'foo', 'name': 'bar'}))
```
Comment (kennywlo) [inline tests/testComposite.py]: PEP8: line too long.

File: tests/testComposite.py (hunk 1 above)
Comment (n8pease) [inline tests/testComposite.py]: < 110.

File: tests/testComposite.py (hunk 2 above)
Comment (n8pease) [inline tests/testComposite.py]: < 110

**Pull request merged**